mlrun/datastore/sources.py

```diff
@@ -1089,9 +1089,10 @@ def __init__(
             attributes["partitions"] = partitions
         sasl = attributes.pop("sasl", {})
         if sasl_user and sasl_pass:
-            sasl["enabled"] = True
+            sasl["enable"] = True
             sasl["user"] = sasl_user
             sasl["password"] = sasl_pass
+            sasl["mechanism"] = sasl.get("mechanism", "PLAIN")
```

Comment:
Since we are here, is it always in upper case? Maybe we can do `sasl.get("mechanism", "PLAIN").upper()`?

Reply:
Actually, I think that in this case it must be `PLAIN` since it's a username-password authentication.
For other mechanisms such as `OAuth` / `SHA`,  are these values aren't relevant.
I'll update.